homeassistant/components/smappee_official/switch.py

```diff
@@ -0,0 +1,197 @@
+"""Support for interacting with Smappee Comport Plugs, Switches and Output Modules."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import BASE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SWITCH_PREFIX = "Switch"
+ICON = "mdi:toggle-switch"
+SCAN_INTERVAL = timedelta(seconds=2)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Smappee Comfort Plugs."""
+    smappee_base = hass.data[DOMAIN][BASE]
+
+    dev = []
+    for _, service_location in smappee_base.smappee.service_locations.items():
+        for actuator_id, actuator in service_location.actuators.items():
+            if actuator.type in ["SWITCH", "COMFORT_PLUG"]:
+                dev.append(
+                    SmappeeActuator(
+                        smappee_base,
+                        service_location,
+                        actuator.name,
+                        actuator_id,
+                        actuator.type,
+                        actuator.serialnumber,
+                    )
+                )
+            elif actuator.type == "INFINITY_OUTPUT_MODULE":
+                for option in actuator.state_options:
+                    dev.append(
+                        SmappeeActuator(
+                            smappee_base,
+                            service_location,
+                            actuator.name,
+                            actuator_id,
+                            actuator.type,
+                            actuator.serialnumber,
+                            actuator_state_option=option,
+                        )
+                    )
+
+    async_add_entities(dev, True)
+
+
+class SmappeeActuator(SwitchEntity):
+    """Representation of a Smappee Comport Plug."""
+
+    def __init__(
+        self,
+        smappee_base,
+        service_location,
+        name,
+        actuator_id,
+        actuator_type,
+        actuator_serialnumber,
+        actuator_state_option=None,
+    ):
+        """Initialize a new Smappee Comfort Plug."""
+        self._smappee_base = smappee_base
+        self._service_location = service_location
+        self._actuator_name = name
+        self._actuator_id = actuator_id
+        self._actuator_type = actuator_type
+        self._actuator_serialnumber = actuator_serialnumber
+        self._actuator_state_option = actuator_state_option
+        self._state = self._service_location.actuators.get(actuator_id).state
+        self._connection_state = self._service_location.actuators.get(
+            actuator_id
+        ).connection_state
+
+        async def async_refresh(event_time):
+            await self.async_update()
+
+        async_track_time_interval(self._smappee_base.hass, async_refresh, SCAN_INTERVAL)
```

Comment:
Removed the custom update helper. It was there in the first place because our switch is actually both a poll/push entity (api/mqtt), but no longer needed by the MIN_TIME_BETWEEN_UPDATES interval constant which has been set quite low.